src/fmt.zig

```diff
@@ -227,7 +243,18 @@ fn formatExpr(fmt: *Formatter, ei: ExprIdx) void {
         .lambda => |l| {
             fmt.push('|');
             var i: usize = 0;
-            for (fmt.ast.store.patternSlice(l.args)) |arg| {
+            const arg_slice = fmt.ast.store.patternSlice(l.args);
+
+            // TODO -- this is a hack to avoid ambiguity with no arguments,
+            // if we parse it again without the space it will be parsed as
+            // a logical OR `||` instead
+            //
+            // desired behaviour described here https://roc.zulipchat.com/#narrow/channel/395097-compiler-development/topic/zig.20compiler.20-.20spike/near/504453049
```

Comment:
Leaving this for someone to do in a follow up PR, for now we're keeping the fuzzer happy

Reply:
Yeah sounds like the correct fix is minor and in the tokenizer. Just remove `||` and `&&` parsing. Then it is not ambiguous.

Reply:
We've merged the changes into the tokeniser. My preference is to revert this in a follow up PR and not add more to this.